include/ParseUrl.php

```diff
@@ -0,0 +1,482 @@
+<?php
+
+/**
+ * @file include/ParseUrl.php
+ * @brief Get informations about a given URL
+ */
+
+namespace Friendica;
+
+use \Friendica\Core\Config;
+
+require_once("include/network.php");
+require_once("include/Photo.php");
+require_once("include/oembed.php");
+
+/**
+ * @brief Class with methods for extracting certain content from an url
+ */
+class ParseUrl {
+
+	public static function getSiteinfoCached($url, $no_guessing = false, $do_oembed = true) {
+
+		if ($url == "") {
+			return false;
+		}
+
+		$r = q("SELECT * FROM `parsed_url` WHERE `url` = '%s' AND `guessing` = %d AND `oembed` = %d",
+			dbesc(normalise_link($url)), intval(!$no_guessing), intval($do_oembed));
+
+		if ($r) {
+			$data = $r[0]["content"];
+		}
+
+		if (!is_null($data)) {
+			$data = unserialize($data);
+			return $data;
+		}
+
+		$data = self::getSiteinfo($url, $no_guessing, $do_oembed);
+
+		q("INSERT INTO `parsed_url` (`url`, `guessing`, `oembed`, `content`, `created`) VALUES ('%s', %d, %d, '%s', '%s')
+			 ON DUPLICATE KEY UPDATE `content` = '%s', `created` = '%s'",
+			dbesc(normalise_link($url)), intval(!$no_guessing), intval($do_oembed),
+			dbesc(serialize($data)), dbesc(datetime_convert()),
+			dbesc(serialize($data)), dbesc(datetime_convert()));
+
+		return $data;
+	}
+
+	public static function getSiteinfo($url, $no_guessing = false, $do_oembed = true, $count = 1) {
```

Comment:
Have a look at the note in the PR description. This method (also some others) don't have a description because I was hoping that @annando could add it after accepting. This is because of the fact that (especially this method) does need some more explanation about its logic. (Where does it scrape which element?) Also some arguments are unknown to me. (What is `$no_guessing`? What does `$do_oembed` do because it is only used for `links` and `photo` and not for `rich` or other content.) There is also some interconnections with `getSiteinfo()`, `getSiteinfoCached()` and `oembed_fetch_url()` (`getSiteinfo()` can call `oembed_fetch_url()` which can call `getSiteinfoCached()` which can call `getSiteinfo()` - I don't understand it completely - but it should be part of the docu). It would be really nice if someone who is more familiar with this code can do this.